posthog/async_migrations/runner.py

```diff
@@ -222,19 +209,19 @@ def run_next_migration(candidate: str):
         trigger_migration(migration_instance)
 
 
-def is_migration_dependency_fulfilled(migration_name: str) -> Tuple[bool, Optional[str]]:
+def is_migration_dependency_fulfilled(migration_name: str) -> Tuple[bool, str]:
     dependency = get_async_migration_dependency(migration_name)
 
     dependency_ok: bool = (
         not dependency or AsyncMigration.objects.get(name=dependency).status == MigrationStatus.CompletedSuccessfully
     )
-    error = f"Could not trigger migration because it depends on {dependency}" if not dependency_ok else None
+    error = f"Could not trigger migration because it depends on {dependency}" if not dependency_ok else ""
```

Comment:
Why `""` vs `None`? Seems not optimal

Reply:
problem: 
this function returns 2 things: ok:bool, error: error str if there was an error (initially `Optional[str]`, proposing `str` type)
If we look at how it's used:
```
ok, error = fn()
if not ok:
  process_error(error)
  
I want `process_error` to require `error:str` as that's the intuitive thing, one shouldn't be able to pass `None` there.
typechecker will complain if we try to pass it somewhere where we expect it to be a string and not None, it seemed more intuitive to change it in this function rather than at all usage locations needing to deal with the typechecker not knowing that if `ok` is `false` then `error` is `:str`. Couldn't figure out a better solution. Ideally it would be cool if we just returned None for no-error and str for the error, but couldn't figure out the fn name for that case 😅